src/gpsea/preprocessing/_patient.py

```diff
@@ -38,13 +38,20 @@ def __init__(self, patient_creator: PatientCreator[T]):
 
     def process(self, inputs: typing.Iterable[T], notepad: Notepad) -> Cohort:
         patients = []
+        patient_ids = set()
+        duplicate_pat_ids = set()
 
         for i, pp in enumerate(inputs):
             sub = notepad.add_subsection(f'patient #{i}')
             patient = self._pc.process(pp, sub)
+            if patient.patient_id in patient_ids:
+                duplicate_pat_ids.add(patient.patient_id)
```

Comment:
I think it is a little better to use `labels` instead of `patient_id` because `labels` is the primary data type.

`SampleLabels` has `__eq__` and `__hash__` so we can check if the items are unique.